scheduler/src/cook/kubernetes/compute_cluster.clj

```diff
@@ -349,23 +305,41 @@
           (let [using-pools? (config/default-pool)
                 synthetic-task-pool-name (when using-pools? pool-name)
                 new-task-requests (remove (fn [{{:keys [job/uuid]} :job}]
-                                            (some #(= (str uuid) (controller/synthetic-pod->job-uuid %))
+                                            (some #(= (str uuid) (synthetic-pod->job-uuid %))
                                                   outstanding-synthetic-pods))
                                           task-requests)
-                task-metadata-seq (->>
-                                    new-task-requests
-                                    (map (fn [{{:keys [job/uuid] :as job} :job}]
-                                           {:task-id (str "synthetic-" pool-name "-" (d/squuid))
-                                            :command {:user user :value command}
-                                            :container {:docker {:image image}}
-                                            :task-request {:scalar-requests (walk/stringify-keys
-                                                                              (tools/job-ent->resources job))
-                                                           :job {:job/pool {:pool/name synthetic-task-pool-name}}}
-                                            ; We need to label the synthetic pods so that we
-                                            ; can opt them out of some of the normal plumbing,
-                                            ; like mapping status back to a job instance
-                                            :pod-labels {api/cook-synthetic-pod-job-uuid-label (str uuid)}}))
-                                    (take (- max-pods-outstanding num-synthetic-pods)))]
+                sidecar-resource-requirements (-> (config/kubernetes) :sidecar :resource-requirements)
+                task-metadata-seq
+                (->> new-task-requests
+                     (map (fn [{{:keys [job/uuid] :as job} :job}]
+                            {:task-id (str api/cook-synthetic-pod-name-prefix "-" pool-name "-" uuid)
```

Comment:
### Why?

Using the job UUID in the synthetic pod name makes tracing these much nicer than having to inspect the pod label.

Reply:
Yes, we're checking for that a few lines above:

```
                new-task-requests (remove (fn [{{:keys [job/uuid]} :job}]
                                            (some #(= (str uuid) (synthetic-pod->job-uuid %))
                                                  outstanding-synthetic-pods))
                                          task-requests)
```

If for some reason this check fails, it's actually a feature that this will prevent us from over-submitting synthetic pods.